tests/shouldwork/LoadModules/BlockRam/DualBlockRam.hs

```diff
@@ -0,0 +1,95 @@
+import Clash.Explicit.Testbench
+import Clash.Sized.Internal.BitVector
+import DualBlockRamDefinitions
+import Test.Tasty.Clash.CollectSimResults
+
+testBench  wmA wmB rstA rstB simOutA simOutB =
+ strictAnd <$> doneA <*> (unsafeSynchronizer clkB clkA doneB)
+  where
+    --topEntity output
+    (portA, portB) = topOut clkA clkB wmA wmB rstA rstB
+    actualOutputA = ignoreFor clkA rstA enableGen d1 undefined portA
+    actualOutputB = ignoreFor clkB rstB enableGen d1 undefined portB
+
+    --Verification
+    outputVerifierA = outputVerifierWith
+     (\clk rst -> assertBitVector clk rst "outputVerifierBitVector Port A")
+    outoutVerifierB = outputVerifierWith
+     (\clk rst -> assertBitVector clk rst "outputVerifierBitVector Port B")
+
+    doneA  = outputVerifierA clkA rstA simOutA $ pack <$> actualOutputA
+    doneA' = not <$> doneA
+    doneB  = outoutVerifierB clkB rstB simOutB $ pack <$> actualOutputB
+    doneB' = not <$> doneB
+
+    (clkA, clkB) = (tbClockGen doneA', tbClockGen doneB')
+    -- Testbench clocks
+
+-- testBench  configurations
+testBench_WF_WF_AB = testBench WriteFirst WriteFirst rst20 rst10 simOutA_WF_WF_AB simOutB_WF_WF_AB
+testBench_WF_WF_BC = testBench WriteFirst WriteFirst rst10 rst7 simOutA_WF_WF_BC simOutB_WF_WF_BC
+testBench_WF_RF_AB = testBench WriteFirst ReadFirst rst20 rst10 simOutA_WF_RF_AB simOutB_WF_RF_AB
+testBench_WF_RF_BC = testBench WriteFirst ReadFirst rst10 rst7 simOutA_WF_RF_BC simOutB_WF_RF_BC
+testBench_WF_NC_AB = testBench WriteFirst NoChange rst20 rst10 simOutA_WF_NC_AB simOutB_WF_NC_AB
+testBench_WF_NC_BC = testBench WriteFirst NoChange rst10 rst7 simOutA_WF_NC_BC simOutB_WF_NC_BC
+testBench_RF_WF_AB = testBench ReadFirst WriteFirst rst20 rst10 simOutA_RF_WF_AB simOutB_RF_WF_AB
+testBench_RF_WF_BC = testBench ReadFirst WriteFirst rst10 rst7 simOutA_RF_WF_BC simOutB_RF_WF_BC
+testBench_RF_RF_AB = testBench ReadFirst ReadFirst rst20 rst10 simOutA_RF_RF_AB simOutB_RF_RF_AB
+testBench_RF_RF_BC = testBench ReadFirst ReadFirst rst10 rst7 simOutA_RF_RF_BC simOutB_RF_RF_BC
+testBench_RF_NC_AB = testBench ReadFirst NoChange rst20 rst10 simOutA_RF_NC_AB simOutB_RF_NC_AB
+testBench_RF_NC_BC = testBench ReadFirst NoChange rst10 rst7 simOutA_RF_NC_BC simOutB_RF_NC_BC
+testBench_NC_WF_AB = testBench NoChange WriteFirst rst20 rst10 simOutA_NC_WF_AB simOutB_NC_WF_AB
+testBench_NC_WF_BC = testBench NoChange WriteFirst rst10 rst7 simOutA_NC_WF_BC simOutB_NC_WF_BC
+testBench_NC_RF_AB = testBench NoChange ReadFirst rst20 rst10 simOutA_NC_RF_AB simOutB_NC_RF_AB
+testBench_NC_RF_BC = testBench NoChange ReadFirst rst10 rst7 simOutA_NC_RF_BC simOutB_NC_RF_BC
+testBench_NC_NC_AB = testBench NoChange NoChange rst20 rst10 simOutA_NC_NC_AB simOutB_NC_NC_AB
+testBench_NC_NC_BC = testBench NoChange NoChange rst10 rst7 simOutA_NC_NC_BC simOutB_NC_NC_BC
+
+-- Expected output generation for all configurations with Template Haskell
+processSimOutput x = replace (0 ::Integer) undefined# $ tail x
```

Comment:
Ah no, I also needed an additional fix in PR #2081: I no longer reset `outputVerifier` to get it running right from the get go. You also enable the reset on `outputVerifier` again. It's almost as if you took an old version of the code rather than the version that is in `master`. Wait... that's it, isn't it? You took an "ours" merge strategy on that file!